src/realm-settings/KeysProvidersTab.tsx

```diff
@@ -79,10 +79,12 @@ export const KeysTabInner = ({ components, refresh }: KeysTabInnerProps) => {
     "org.keycloak.keys.KeyProvider"
   ].map((item) => item.id);
 
-  const itemIds = components.map((_, idx) => "data" + idx);
+  const itemIds = components.map((component) => component.id);
```

Comment:
This is probably one of those cases where a model from Keycloak is missing type information about whether the field is actually optional. You might be able to use a non-null assertion here to get rid of the undefined values.
```suggestion
  const itemIds = components.map((component) => component.id!);
```